video/out/w32_common.c

```diff
@@ -1543,6 +1582,10 @@ static int gui_thread_control(struct vo_w32_state *w32, int request, void *arg)
         if (w32->opts->fullscreen != w32->current_fs)
             reinit_window_state(w32);
         return VO_TRUE;
+    case VOCTRL_GET_FULLSCREEN:
+        *(int *)arg = w32->toggle_fs ^ w32->current_fs;
+        reinit_window_state(w32);
```

Comment:
The only way I see is to move `reinit_window_state` above the message queue handler and call right after the `signal_events`. Does that sound okay?